### PR TITLE
Add fast path of qmean/qstd for quantized CPU

### PR DESCRIPTION
**Pull request by Xia-Weiwen**
Add fast path of qmean and qstd when computation is done in innermost dimensions for quantized CPU. The fast path supports inputs in contiguous memory format.
For example:
```python
X = torch.randn((2,3,4,5), dtype=torch.float)
qX = torch.quantize_per_tensor(X, scale, zero_point, torch_type)

# dim can be: -1, (-1, -2), (-1, -2, -3), (-1, -2, -3, -4), 3, (3, 2), (3, 2, 1), (3, 2, 1, 0) or None
dim = -1
qY = torch.mean(qX, dim) # qY = torch.std(qX, dim)
```

**Performance test results**
Test Env:
- Intel® Xeon® CLX-8260
- 1 instance, 4 cores
- Using Jemalloc

Test method:
Create 4d contiguous tensors as inputs, set `dim` to the innermost two dimensions `(-1, -2)`, then do the following tests
- Quantize inputs and use the fast path
- Quantize inputs and use the reference path
- Use fp32 kernel (no quantization)

Mean: exec time (us) vs. shape
image

Std: exec time (us) vs. shape
image

### CODE REVIEW

File: test/quantization/core/test_quantized_op.py [hunk 1]
```diff
@@ -1903,6 +1909,31 @@ def test_quantized_mean_qnnpack(self, keep):
             MQ = XQ.mean((2, 3), keepdim=keep)
             self.assertTrue(torch.equal(MQ, YQ))
 
+    @given(X=hu.tensor(shapes=hu.array_shapes(min_dims=2, max_dims=5,
```
Comment (vkuzo) [inline test/quantization/core/test_quantized_op.py:1912]: Although a lot of existing quantization tests use hypothesis for test case generation, this is now discouraged because the values hypothesis creates are not representative of real tensors.  Any chance we could switch the test to create the values without using hypothesis?  Here is an example: https://github.com/pytorch/pytorch/blob/master/test/quantization/core/test_quantized_op.py#L268

Unfortunately with the current setup, this test may not always catch any issues with the actual calculations for representative values inside the tensors.

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp [hunk 2]
```diff
@@ -2602,6 +2602,119 @@ void quantized_normalize_kernel(
...
+  auto x_scale = self.q_scale();
+  auto x_zp = self.q_zero_point();
+  result = at::_empty_affine_quantized(
+      out_dims,
+      at::device(kCPU).dtype(dtype).memory_format(self.suggest_memory_format()),
+      x_scale,
+      x_zp,
+      c10::nullopt);
+
+  AT_DISPATCH_QINT_TYPES(self.scalar_type(), "quantized_layer_norm_kernel_impl_cpu", [&]() {
```
Comment (vkuzo) [inline aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:2692]: nit: "quantized_layer_norm_kernel_impl_cpu" to "quantized_qstd_kernel_impl_cpu"?

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp [hunk 3]
```diff
@@ -2602,6 +2602,119 @@ void quantized_normalize_kernel(
...
+    out_dims.erase(out_dims.end() - num_dims_to_squeeze, out_dims.end());
+  }
+  result = at::_empty_affine_quantized(
+      out_dims,
+      at::device(kCPU).dtype(dtype).memory_format(self.suggest_memory_format()),
+      self.q_scale(),
+      self.q_zero_point(),
+      c10::nullopt);
+
+  AT_DISPATCH_QINT_TYPES(self.scalar_type(), "quantized_layer_norm_kernel_impl_cpu", [&]() {
```
Comment (vkuzo) [inline aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:2637]: nit: "quantized_layer_norm_kernel_impl_cpu" to "quantized_qmean_kernel_impl_cpu"?

**Review by vkuzo — COMMENTED** [review]
thanks for the improvement!  I had one comment about generating test tensors without hypothesis, other than that looks great.

**Review by malfet — APPROVED** [review]

**Review by malfet — CHANGES_REQUESTED** [review]
Please make sure that @vkuzo comments are addressed

File: test/quantization/core/test_quantized_op.py (hunk 1 above)
Comment (Xia-Weiwen) [reply test/quantization/core/test_quantized_op.py]: Thanks for the suggestion. I have reimplemented the test with `itertools.product`.

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp (hunk 2 above)
Comment (Xia-Weiwen) [reply aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp]: Thanks. It is now fixed.

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp (hunk 3 above)
Comment (Xia-Weiwen) [reply aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp]: Thanks. It is now fixed.

**Comment by Xia-Weiwen** [comment]
Hi @malfet I have rebased this PR and those issues mentioned by @vkuzo have been fixed. Please review again. Thanks!

**Review by malfet — APPROVED** [review]

**Comment by malfet** [comment]
@pytorch merge -f

**Comment by malfet** [comment]
@suo  you beat me to it by a sec :)
